torch/csrc/dynamo/eval_frame.c

```diff
@@ -724,16 +775,18 @@ static PyObject* _custom_eval_frame(
   }
   // cache miss
 
-  PyObject *frame_state = get_frame_state(frame->f_code);
+  PyObject *frame_state = NULL;
+  if (extra != NULL) {
+    frame_state = extra->frame_state;
+  }
   if (frame_state == NULL) {
     // TODO(voz): Replace this dict with a real FrameState object.
     frame_state = PyDict_New();
-    set_frame_state(frame->f_code, frame_state);
```

Comment:
I think you leak `frame_state` now when you create a skip.

Reply:
Very nice catch!